misc/config_tools/launch_config/launch_cfg_gen.py

```diff
@@ -165,6 +168,8 @@ def add_virtual_device(self, kind, vbdf=None, options=""):
 
         if vbdf is None:
             vbdf = self._vbdf_allocator.get_virtual_bdf()
+        else:
+            self._vbdf_allocator.remove_virtual_bdf(vbdf)
```

Comment:
If you choose to remove here, drop a comment in `generate_for_one_vm` highlighting that ivshmem and vuart must be the first virtual devices generated before the others except hostbridge and LPC which have their own reserved slots.